File: year2022/day01/java/Solution.java
import java.nio.file.*;
import java.util.*;

public class Solution {
    private static List<Integer> ELVES = new ArrayList<Integer>();
    
    public static void main(String[] args) { 
        try {
            setup(Files.readAllLines(Paths.get("../puzzle.txt")));
            System.out.println(part1());
            System.out.println(part2());
        } catch(Exception e) {
            e.printStackTrace();
        }
    }
    
	public static Integer part1() {
        return Collections.max(ELVES);
    }
    
    public static Integer part2() {
        return ELVES.stream().sorted(Collections.reverseOrder())
                .limit(3).mapToInt(Integer::intValue).sum();
    }
    
    public static void setup(List<String> input) {
        int sum = 0;
        for(String calory : input) {
            if(calory.isBlank()) {
                ELVES.add(sum);
                sum = 0;
            } else {
                sum += Integer.parseInt(calory);
            }
        }
    }
}

File: year2022/day01/python/solution.py
def main():
    print(part_1())
    print(part_2())


def part_1():
    return max(input)


def part_2():
    return sum(sorted(input, reverse=True)[:3])


def setup(path):
    global input
    
    with open(path, 'r') as file:
        input = [sum(map(int, input_string.split())) for input_string in file.read().split('\n\n')]


if __name__ == '__main__':
    setup('../puzzle.txt')
    main()